### PR TITLE
Add copy method to mesher classes

### LINKED ISSUE #284: Add copy method to classes in mesher
Sometimes I'll need to make copies of `mesher` objects. It would be great to have a `.copy()` method in all classes.

### PR DESCRIPTION
**Pull request by PiotrKurnik**
Fixes #284 

I added copy method to Square. Also I wrote some tests. This is for a try, if OK I will make further changes.

### Checklist:

- [x] Make tests for new code (at least 80% coverage)
- [x] Create/update docstrings
- [x] Include relevant equations and citations in docstrings
- [x] Docstrings follow the style conventions
- [x] Code follows PEP8 style conventions
- [x] Code and docs have been spellchecked
- [x] Include new dependencies in `doc/install.rst`, `requirements.txt`, `environment.yml`, `ci/requirements-conda.txt` and `ci/requirements-pip.txt`.
- [x] Documentation builds properly (run `cd doc; make` locally)
- [x] Changelog entry (leave for last to avoid conflicts)
- [x] Firt-time contributor? Add yourself to `doc/contributors.rst` (leave for last to avoid conflicts)

### CODE REVIEW

**Comment by leouieda** [comment]
Thank you for doing this! The code and tests are looking pretty good! I don't think there is much else to do but a few minor changes and documentation.

The reason Travis CI is failing is because of the style check (PEP8). See [here](https://travis-ci.org/fatiando/fatiando/jobs/145380872#L1195) for the error log. You just have one too many lines after the copy method. You can run `make pep8` locally to check this.

One other thing I would ask is that you rename your test file to `test_square.py` because it will contain only the tests for Square.

File: fatiando/tests/mesher/test_mesher.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import numpy as np
+from ...mesher import Square
+
+
+def test_square_copy():
+    orig = Square([0, 1, 2, 4], {'density': 750})
+    cp = orig.copy()
```
Comment (leouieda) [inline fatiando/tests/mesher/test_mesher.py]: Another check you do is to check is an object is the same reference as another using the `is` comparison: `cp is not orig`. 

Two objects might be equal but not be the same object.

**Comment by leouieda** [comment]
@PiotrKurnik let me know when you make some of these changes so that I can take a look. No pressure and no hurry :smile: 

One things I forgot to mention is that the `copy` method needs a docstring. Please try to follow the style of the others. See http://www.fatiando.org/dev/develop.html#documentation for a specification.

**Comment by PiotrKurnik** [comment]
@leouieda I have added `copy` to other classes and I have written  a few  more tests. It's not finished yet, just wanted to see if Travis will pass the tests. It shouldn't take long to complete this.

**Comment by leouieda** [comment]
@PiotrKurnik OK! Let me know when you're finished and I'll review the code to see if there is anything else needed.

Great job on this!

File: fatiando/tests/mesher/test_prism_mesh.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from ...mesher import PrismMesh
+
+
+def test_prism_mesh_copy():
+    p1 = PrismMesh((0, 1, 0, 2, 0, 3), (1, 2, 2))
+    p1.addprop('density', 3200)
```
Comment (leouieda) [inline fatiando/tests/mesher/test_prism_mesh.py]: For the mesh classes in Fatiando, the physical properties are arrays of values (one for each cell in the mesh). Could you test this using arrays? You can create the array with something like `p1.addprop('density', 3200 + np.zeros(p1.size))`

**Comment by leouieda** [comment]
@PiotrKurnik one thing I  just remembered, all basic geometric types (`Square`, `Prism` etc but not `PrismMesh` etc) inherit from `GeometricElemennt`. So you can implement the `copy` method there and they will all  get it through inheritance. Sorry it took me so long to spot this. I've haven't looked at the mesher code for quite some time.

**Comment by PiotrKurnik** [comment]
@leouieda Good spot! I have pushed the code. Let's see if it's any closer now.

File: fatiando/mesher.py [hunk 3]
```diff
@@ -122,6 +127,10 @@ def x(self):
     def y(self):
         return self.vertices[:, 1]
 
+    def copy(self):
+        """ Return a deep copy of the current instance."""
+        return cp.deepcopy(self)
+
```
Comment (leouieda) [inline fatiando/mesher.py]: You can remove this one since `Polygon` will inherit it from `GeometricElement`.

File: doc/changelog.rst [hunk 4]
```diff
@@ -9,11 +9,12 @@ Version 0.5
 **Release date**: yyyy-mm-dd
 
 **Changes**:
-
+* Add copy method to fatiando.mesher.
```
Comment (leouieda) [inline doc/changelog.rst]: Could you please change this to the following?

    Add a copy method to ``fatiando.mesher`` objects.

File: doc/changelog.rst [hunk 5]
```diff
@@ -9,11 +9,12 @@ Version 0.5
...
+  ('PR 301  <https://github.com/fatiando/fatiando/pull/301>`__)
```
Comment (leouieda) [inline doc/changelog.rst]: The first `'` should be a back tic like the closing one.

**Comment by leouieda** [comment]
@PiotrKurnik all looks good! Just made a few minor comments. Once those are done let me know and I'll hapilly merge this. Thank you for putting time into this!

**Comment by PiotrKurnik** [comment]
Thanks @leouieda. I am sorry it took so long.

File: doc/contributors.rst [hunk 6]
```diff
@@ -14,6 +14,7 @@ A (hopefully) updated list of people who contributed to Fatiando a Terra:
 * `Martin Bentley`_ - Nelson Mandela Metropolitan University, South Africa and AEON, South Africa
 * `Victor Almeida`_ - UERJ, Brazil.
 * `M. Andy Kass`_ - United States Geological Survey, USA
+* `Piotr Kurnik`_ - United Kingdom
```
Comment (leouieda) [inline doc/contributors.rst]: Do you want to link your name to some page (a personal page, github profile, etc)? If so, add the link to the lines below like the other contributors. If not, you can remove the back tics and the underline.

Sorry that I didn't see this before.

**Comment by leouieda** [comment]
@PiotrKurnik no problem! We're all busy with other work and I'm the last one that can complain.

File: doc/contributors.rst (hunk 6 above)
Comment (PiotrKurnik) [inline doc/contributors.rst]: I removed the back ticks.

File: doc/changelog.rst [hunk 7]
```diff
@@ -10,10 +10,12 @@ Version 0.5
 
 **Changes**:
 
+* Add copy method to ``fatiando.mesher`` objects.
+  (`PR 301  <https://github.com/fatiando/fatiando/pull/301>`__)
 * Move from ``distutils`` to ``setuptools`` in ``setup.py``, as recommended in
   the `Python Packaging User Guide <https://packaging.python.org/>`__.
   (`PR 294 <https://github.com/fatiando/fatiando/pull/294>`__)
-* Enable ``fatiando.mesher.PointGrid`` to have points at different depths by
+* Enable ``fatiando.mesher.PointGrid`a` to have points at different depths by
```
Comment (leouieda) [inline doc/changelog.rst]: Looks like an `a` was inserted here by mistake. Could you please delete it?

**Comment by leouieda** [comment]
@PiotrKurnik everything builds properly on my machine and we're ready to merge!

File: doc/changelog.rst (hunk 7 above)
Comment (PiotrKurnik) [inline doc/changelog.rst]: no idea where it came from ....

**Comment by leouieda** [comment]
@PiotrKurnik merged! Congratulations on your first contribution (of many I hope)!

**Pull request merged**